From f579336438ae5827d84dbc5d54b77da7abcebcfa Mon Sep 17 00:00:00 2001
From: Michael Webster <miketwebster@gmail.com>
Date: Tue, 10 Jan 2017 12:26:38 -0500
Subject: [PATCH] stage.py: clear clipboards only once, when waking. 
 raise_unlock_widget gets called repeatedly when there is input (to refresh
 other things) - having it repeatedly clear the clipboard was unnecessary, and
 prevented keyboard navigation actions that included selection of characters -
 such as ctrl-a, shift-home and -end.

Fixes #197
---
 src/stage.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

Index: cinnamon-screensaver/src/stage.py
===================================================================
--- cinnamon-screensaver.orig/src/stage.py	2017-04-02 11:46:12.793587440 +0200
+++ cinnamon-screensaver/src/stage.py	2017-04-02 11:46:12.789587337 +0200
@@ -466,11 +466,12 @@
         our other reveals after its transition is complete.
         """
         self.reset_timeout()
-        utils.clear_clipboards(self.unlock_dialog)
 
         if status.Awake:
             return
 
+        utils.clear_clipboards(self.unlock_dialog)
+
         self.clock_widget.stop_positioning()
         self.albumart_widget.stop_positioning()
 
